python/bolt-mcp/src/bolt_mcp/transport.py

```
@@ -0,0 +1,390 @@
+    with contextlib.suppress(Exception):
+        request.state["context"] = {
+            "user_id": claims.get("sub"),
+            "is_staff": bool(claims.get("is_staff")),
+            "is_superuser": bool(claims.get("is_superuser")),
+            "permissions": claims.get("permissions") or [],
+            "auth_claims": claims,
+        }
+    return None
+
+
+def _resolve_oauth(api: BoltAPI, oauth: Any) -> ProtectedResource | None:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `oauth`

(ANN401)